src/Avalonia.Controls/TickBar.cs

```diff
@@ -0,0 +1,430 @@
+using System;
+using System.Collections.Generic;
+using Avalonia.Controls.Primitives;
+using Avalonia.Data;
+using Avalonia.Data.Converters;
+using Avalonia.Layout;
+using Avalonia.Media;
+using Avalonia.Utilities;
+
+namespace Avalonia.Controls
+{
+
+    /// <summary>
+    /// Enum which describes how to position the TickBar.
+    /// </summary>
+    public enum TickBarPlacement
+    {
+        /// <summary>
+        /// Position this tick at the left of target element.
+        /// </summary>
+        Left,
+
+        /// <summary>
+        /// Position this tick at the top of target element.
+        /// </summary>
+        Top,
+
+        /// <summary>
+        /// Position this tick at the right of target element.
+        /// </summary>
+        Right,
+
+        /// <summary>
+        /// Position this tick at the bottom of target element.
+        /// </summary>
+        Bottom,
+
+        // NOTE: if you add or remove any values in this enum, be sure to update TickBar.IsValidTickBarPlacement()
+    }
+
+    /// <summary>
+    /// TickBar is an element that use for drawing Slider's Ticks.
+    /// </summary>
+    public class TickBar : TemplatedControl
```

Comment:
```suggestion
    public class TickBar : Control
```

There is not template here and in WPF it is not a template based control either.

Reply:
agreed, i used this to test if it was receiving template events.. apparently not :)